dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/NativeResource.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.dlr.jni;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code NativeResource} is an internal class for {@link AutoCloseable} blocks of memory created in
+ * the DLR Engine.
+ */
+public abstract class NativeResource implements AutoCloseable {
+
+    private static final Logger logger = LoggerFactory.getLogger(NativeResource.class);
+
+    protected AtomicLong handle;
+    private Exception exception;
+
+    protected NativeResource(long handle) {
+        this.handle = new AtomicLong(handle);
+        if (logger.isTraceEnabled()) {
+            exception = new Exception();
+        }
+    }
+
+    /**
+     * Gets the boolean that indicates whether this resource has been released.
+     *
+     * @return whether this resource has been released
+     */
+    public boolean isReleased() {
+        return handle == null;
```

Comment:
handler should never be null
```suggestion
        return handle.get() == null;
```